tests/Integration/Behaviour/Features/Context/ProductFeatureContext.php

```diff
@@ -83,13 +85,39 @@ public function before(BeforeScenarioScope $scope)
     /* PRODUCTS */
 
     /**
-     * @param $productName
+     * @param string $productName
      *
      * @return Product
      */
-    public function getProductWithName($productName)
+    public function getProduct(string $productName)
+    {
+        $idShop = (int) Context::getContext()->shop->id !== (int) Configuration::get('PS_SHOP_DEFAULT') ?
```

Comment:
Mmmm, stateful tests to validate stateful crappy code ^^
What are the chances this goes wrong someday? :trollface: 

(I don't have a better/faster alternative though ^^)

Reply:
<3